### ISSUE TITLE
TRACK: Sprint review 19.3.24

### ISSUE THREAD

**Issue by cmarooney-stfc** [pace-neutrons/Horace#1610]
1. Detailed view of Horace-4 updates/blockers from Ross/Excitations

- Good
   -  5x speedup, may reduce need for immediate profiling, res/conv speedup similar, may reduce need for immediate refactoring. Noting this speedup probably in parallel with 8 cores vs serial

- Bad (see below for more detailed breakdown)
   -    spherical projection (thought to be broken). No specific instances reported - Ross to follow up
   -    code stability - Excitations are finding the use of master as release source problematic as code changes unpredictably

Full report from Ross:

BETTER
- 5x quicker (parallel vs serial, 8 cores)
- full symmetrisation post gen_sqw
- file-backed large cuts
- res/conv slightly quicker than 5x, uses less memory on bigger cuts

WORSE
- backward compatibility (refine crystal, cuts, ....) - should be backwards compatible  apart from alignment where interface changed, old style will issue warnings which may intimidate/be seen as errors. Keep warnings as old interface may be deprecated. NB can disable warnings. Also there is uoffset issue, there is ticket. Ideas to solve this, not yet done. Could set offset=uoffset and warn on imperfect solution. DO NOW
- PR open (JW) how to convert between 3 and 4 (instructions) mostly compatible bar warnings we dive instructions for reducing warnings
- stability (release direct from master, as above)
- error messages from each worker - add verbosity flag to parallel to prevent this or dump to log plus print headnode DO NOW dump to log file (errors) JW to do by Tuesday
- harder to use for "power users" interacting directly with sqw objects - work with Radu 
- length of time to take cuts (read/write speed on IDAAAS) - addressible in the long term, 4 solutions - #1611 

BROKEN
- parallel fitting not compatible with SpinW - harass Duc POST-CONSTRUCTION
- general symmetrisation - symm sqw does not support symop general due to domain issues - needs linear algebra #1612 POST-CONSTRUCTION but cut-sym supports all transformations JW says cut-sym will work for this + python script to generate symmetries - need to check this works on daaas (installation issues) - someone to ask daaas for this - to run pip install spglib 
- projaxes - maybe misunderstanding of warning, check with Ross- now documented for the conversion
- parallel HPC settings differ between H3 and H4 - new H4 features need new settings, documented, need setup guide FIXING NOW JW to provide this - hours work , available early in cycle JW to chat with Ross
- random initial configuration - also part of setup guide - FIXING NOW but need IDAAAS input e.g. for new machines - also awits documentation
- repeat stability issue here as "github admin -> inclusion of changes"
- spherical projection - check with Christian, Russell, - hopefully ok with new docs and auto parameters- AB to add to JW doc for this

MISSING
- memory allocation for parallel processing - warning waiting on merge of docs MERGED see docs for mem usage
- initial configuration user help AS ABOVE this is the document (nb JW to chat with Ross Monday)
- proper version control (on the fly) =  stability 
- normalised polygons - see mslice feature (is this powder?) is this cut-dnd? feature request and optimisation JW to ticket

HUMAN
- trust - low due to above issues
- inertia - still using H3

PROPOSAL
- ensure spherical projection works 
- get users to use this as way into H4 as H3 does not have this

______

2. Major areas - need to ensure closed or track (continued from last time)

 WHAT MUST WE FINISH BY MAR 31?
 - cylindrical projection - ready to start modulo possibility of postconstruction improvements to spherical
                                      - can roll mod-q axis (current measure inv.A) change scale to hkl,  for spherical, into this task
                                      - issues added #1578 TO BE DONE, estimate 3d
 - detector info - received from Duc modulo spectra to workspace maps missing (mantid crash/glitch) - TO BE DONE Duc still to do, half day work
 - profiling - - TO BE DONE - data from Duc received, comparison in progress. SQW load reduced but may be format conversion - initial timings w/e 9/3, awaiting data from Toby - SOME BUGS DISCOVERD, USER REQUESTS PRIORITIZED
Testing needed to confirm it works and that it provides speedup given possible high i/o
 - Fan IDAAAS performance issue - fix issued, Fan ticket closed. Find current max file size, discuss with Excitations. Import test scripts into our github. Check usability of Alex instructions. Issue is cache usage of memory for file transfer - there are different takes on this. Chris to consider how instructions work for reader
 - Fan IDAAAS issue .- pending state causing crash. Fixed but not reproducible - machine info with Fan - no more feedback, may not be reproducible
 - SLURM cluster unavailable after SL7 purge - JW to contact Fan to reinstate. Noting gen_sqw did not work well with SLURM. Also bulleted document describing stages of usability - still to be contacted re this 

New:

Alex -

 - [ ] #1591 legacy alignment - issue with matrix separation (U/rotation, B/scale to crystal cartesian) - realignment of H3 file may cause problems - BUG, SO TO BE DONE first,  in progress, (check Ross use of 3.6.3) then Cylindrical - 20/3 finish - issues detected may be delayed
 - [ ] #1583 reassign to JW one hour's work - PR open

Jacob -

 - [ ] #1566  PR still open

Previous issues:

Alex -

Jacob - 
 - [ ] #1553 with AB, Jacob to scope moved from above TO BE DONE
 - [ ] #1553 not critical but extends to half Matlab platforms - parallel toolbox not affected, problem Windows only. Prioritise finding out extent of problem (still post-construction) - Jacob to scope this sprint NOT YET SCOPED, TO BE DONE this sprint
moved to JW below - IDAAAS latest 2023a so not affected, scoping continues

Chris - 
 - [ ] #1544 done, moving on to restoring hashes to saveable fields

**Comment by cmarooney-stfc** [comment]
Closed as superseded by #1650. NB Gap in sprint reviews owing to end of funded project plus Easter break and team absences
